ledger/participant-integration-api/src/main/scala/platform/apiserver/meteringreport/HmacSha256.scala

```diff
@@ -57,7 +38,32 @@ object HmacSha256 {
       .fold(deserializationError(s"Failed to deserialize $json", _), identity)
   }
 
-  final case class Key(encoded: Bytes, algorithm: String)
-  implicit val KeyFormat: RootJsonFormat[Key] = jsonFormat2(Key.apply)
+  /** @param scheme  - a long lived name that be associated with, and only with, this key
+    * @param encoded - the encoded bytes of a HmacSha256 secret key
+    * @param algorithm - the key algorithm
+    */
+  final case class Key(scheme: String, encoded: Bytes, algorithm: String)
+
+  implicit val KeyFormat: RootJsonFormat[Key] = jsonFormat3(Key.apply)
+
+  // The key used for both mac and key generation as defined in
+  // https://docs.oracle.com/javase/9/docs/specs/security/standard-names.html
+  val algorithm = "HmacSHA256"
+  private val macPrototype = new MacPrototype(algorithm)
+
+  def compute(key: Key, message: Array[Byte]): Either[Throwable, Array[Byte]] = {
+    Try {
+      val mac = macPrototype.newMac
+      val secretKey = new SecretKeySpec(key.encoded.bytes, key.algorithm)
+      mac.init(secretKey)
+      mac.doFinal(message)
+    }.toEither
+  }
```

Comment:
these braces can be removed, no?

Reply:
Could be but my preference it to brace when there are multiple lines involved (and it isn't _yet_ against the style guidelines) ;-)

Reply:
I changed my mind after Scala3 dropped braces ;) sometimes it just adds noise :)